doc/nrf/glossary.rst

```diff
@@ -0,0 +1,447 @@
+   A controller used for writing and erasing the internal flash memory and the User Information Configuration Registers (UICR).
+
+nRF Connect for Cloud
+   Nordic Semiconductor's platform for connecting IoT devices to the cloud, viewing and analyzing device message data, prototyping ideas that use Nordic Semiconductor chips, and more.
+   It includes a public REST API that can be used for building IoT solutions.
+   See `nRF Connect for Cloud`_.
+
+nRF repository
+   An |NCS| repository that does not have an externally maintained, open-source upstream.
+   It is exclusive to Nordic development.
+
+nRF Secure Immutable Bootloader (NSIB)
+   A bootloader created and maintained by Nordic Semiconductor that is built on Chain of Trust architecture.
+
+OpenAMP
+   A framework that provides software components that enable the development of software applications for Asymmetric Multiprocessing (AMP) systems.
+   See `OpenAMP`_.
+
+OpenThread
+   A portable and flexible `open-source implementation <OpenThread.io>`_ of the Thread networking protocol.
+
+OpenThread Border Router (OTBR)
+   A router that connects a Thread network to other IP-based networks, like Wi-Fi or Ethernet.
+   A Thread network requires a Border Router to connect to other networks.
+
+Operating System (OS)
+   A set of functions and data structures that manages system resources, hardware components, and the execution of programs and processes.
+   It is usually composed of a kernel, a scheduler, a file system, a memory manager, and other components.
+
+OSS repository
+   An |NCS| repository that tracks an upstream Open Source Software counterpart that is externally maintained.
+
+Over-the-Air (OTA)
+   Any type of wireless transmission.
+
+Patch
+   A method to export git commits for sharing them with others.
+   It does not require a repository.
+
+Power Amplifier (PA)
+   A device used to increase the transmit power level of a radio signal.
+
+Power Saving Mode (PSM)
+   A feature introduced in 3GPP Release 12 to improve battery life of IoT (Internet of Things) devices by minimizing energy consumption.
+   The device stays dormant during the PSM window.
+
+Printed Circuit Board (PCB)
+   A board that connects electronic components.
+
+Protocol Data Unit (PDU)
+   Information transferred as a single unit between peer entities of a computer network and containing control and address information or data.
+   PDU mode is one of the two ways of sending and receiving SMS messages.
+
+Provisioning
+   * In a thread mesh network, the process of associating a device to the appropriate service, and performing any application or vendor-specific configuration.
+     It is a step in the commissioning process.
+   * In a bluetooth mesh network, the process of adding devices to the network.
+   * In a bootloader, the process of storing public key hashes in a separate region of memory from the bootloader image.
+
+Pull Request
+   A set of commits that are sent to contribute to a repository.
```

Comment:
This only applies in the context of GitHub

Reply:
@carlescufi could you elaborate? This entry used to be Github-specific, but I suggested to make it more general, as the concept of a pull request is widely used, both on Github and outside of it.

Reply:
@carlescufi

Reply:
OK, I think Pull Request is almost exclusive to GitHub (GitLab, gerrit and others use other terms) but I do think that BitBucket uses it, so I will approve.